modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptNodeClientCodegen.java

```diff
@@ -128,4 +129,67 @@ public String getNpmRepository() {
     public void setNpmRepository(String npmRepository) {
         this.npmRepository = npmRepository;
     }
+
+    @Override
+    public Map<String, Object> postProcessAllModels(Map<String, Object> objs) {
+        Map<String, Object> result = super.postProcessAllModels(objs);
+
+        for (Map.Entry<String, Object> entry : result.entrySet()) {
+            Map<String, Object> inner = (Map<String, Object>) entry.getValue();
+            List<Map<String, Object>> models = (List<Map<String, Object>>) inner.get("models");
+            for (Map<String, Object> mo : models) {
+                CodegenModel cm = (CodegenModel) mo.get("model");
+                // Fix complex array property's datatype
+                for (CodegenProperty prop : cm.allVars) {
+                    if (!prop.isPrimitiveType) {
+                        CodegenModel complexTypeModel = getCodegenModel(result, prop.complexType);
+                        if (complexTypeModel != null && complexTypeModel.isArrayModel) {
+                            prop.complexType = complexTypeModel.arrayModelType;
+                            prop.datatype = complexTypeModel.parent;
+                            prop.baseType = "Array";
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public Map<String, Object> postProcessOperationsWithModels(Map<String, Object> objs, List<Object> allModels) {
+        Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+        if (operations != null) {
+            List<CodegenOperation> ops = (List<CodegenOperation>) operations.get("operation");
+            for (CodegenOperation op : ops) {
+                // Fix complex array object input params of an operation
+                for (CodegenParameter param: op.allParams) {
+                    CodegenModel cm = getCodegenModelFromAllModels(allModels, param.dataType);
+                    if (cm != null && cm.isArrayModel) {
+                        param.dataType = cm.parent;
+                    }
+                }
+                // Fix complex array object return type of an operation
+                CodegenModel cm = getCodegenModelFromAllModels(allModels, op.returnType);
+                if (cm != null && cm.isArrayModel) {
+                    op.returnType = cm.parent;
+                }
+            }
+        }
+        return objs;
+    }
+
+    private CodegenModel getCodegenModelFromAllModels(List<Object> allModels, String modelName) {
+        for (Object o : allModels) {
+            Map<String, Object> mo = (Map<String, Object>) o;
+            CodegenModel cm = (CodegenModel) mo.get("model");
+            if (cm.name.equals(modelName)) {
```

Comment:
It can throw NullPointerException if mo.get("model") return null

Reply:
+1, agree, `NullPointerException` approaching :)

Reply:
+1, agree